src/coreclr/src/jit/emitarm64.cpp

```diff
@@ -4025,6 +4025,11 @@ void emitter::emitIns_R_R(
                 }
             }
 
+            if (IsOppositeOfPrevMov(ins, reg1, reg2))
```

Comment:
Seems like you might need to worry about the EA_4BYTE case here too.

Reply:
Makes sense. I think I will try to move the one above `reg1 == reg2` inside the new `IsOppositeOfPrevMov` so we have all the checks in one place.

Reply:
Yeah, generalize to "is unnecessary mov" and catch all 3 cases:
```
mov   RX, RX

mov RX, RY
mov RY, RX

mov RX, RY
mov RX, RY
```

Reply:
Given that we want to check `OptimizationEnabled()` before doing this optimization and `mov RX, RX` was earlier done without this check, will it be fine to make this optimization only when opts is enabled?